File: gr00t/Isaac-GR00T/examples/H20/robots/sdk_package/example/h10-w/chassis/get_chassis_speed.cpp
// get_chassis_speed.cpp

#include <iostream>
#include <iomanip>
#include "h10-w/h10w_chassis.h"
#include "common/common_def.h"

using namespace humanoid;
using namespace humanoid::h10w;

int main() {
  std::cout << "=== 获取底盘速度示例 ===" << std::endl;
  H10wChassis chassis;
  float linear_velocity;
  float angular_velocity;

  if (chassis.getChassisSpeed(linear_velocity, angular_velocity) == 0) {
    std::cout << "获取成功" << std::endl;
    std::cout << "    线速度=" << linear_velocity << "m/s, 角速度="
              << angular_velocity << "rad/s" << std::endl;
  } else {
    std::cout << "获取失败" << std::endl;
    return -1;
  }
  return 0;
}

File: gr00t/Isaac-GR00T/examples/H20/robots/sdk_package/example_py/h10-w/chassis/get_chassis_speed.py
import humanoid_sdk_py
import humanoid_sdk_py.h10w as h10w

print("=== get chassis speed ===")
h10w_chassis = h10w.H10wChassis()
ret, tuple_data = h10w_chassis.getChassisSpeed()
if ret != 0:
    print("get chassis speed failed.")
    exit(-1)

linear_velocity, angular_velocity = tuple_data
print("get chassis speed success.")
print(
    f"chassis speed: linear velocity:{linear_velocity}, angular velocity:{angular_velocity}"
)
